### ISSUE TITLE
Use cluffer:buffer-s in input-edit-mode

### ISSUE THREAD

**Issue by aartaka** [atlas-engineer/nyxt#1321]
Create full `cluffer:buffer`s, instead of `cluffer:line`s. This way we will make it more performant and simple (I've just tried to implement `cursor-next-line` and was overwhelmed by the complexity of that). After all, `cluffer` distinguishes between `buffer` and `line` for a reason. 

Current implementation is quite slow on long text fields, too.

**Comment by Ambrevar** [comment]
Makes sense to me.
@jmercouris ?

**Comment by jmercouris** [comment]
Yes, makes sense to me too!

**Comment by aadcg** [comment]
Yes, this is indeed a mistake that probably derived from not carefully studying the `cluffer` library. 

Given that we have zillions of ways to have editor-like behaviors in Nyxt (each with its own set of quirks), we should probably firstly understand what we want.
